### ISSUE TITLE
Python 3 support

### ISSUE THREAD

**Issue by espenfl** [conwetlab/ckanext-oauth2#39]
Thanks for this great extension. It seems this is the only one that is maintained for OAuth 2.0, or am I mistaken?

Are there efforts to try to upgrade this to Python 3?

**Comment by aitormagan** [comment]
I am afraid there are no plans to upgrade this extension to Python 3 😞.

**Comment by espenfl** [comment]
@aitormagan Thanks for the reply. Do you know of other maintained extensions for CKAN that offers OAuth 2 connectivity? When you say there are no plans to upgrade to Python 3, is that strictly due to lack of resources to do so, or is there some technicalities that would be difficult to address? If former, we might assist.

**Comment by syats** [comment]
Hi @espenfl  did you find any solution in this respect? 
I am evaluating using CKAN for two projects but carrying around python-2 dependencies is a huge disadvantage.

**Comment by aitormagan** [comment]
@espenfl, I am sorry I didn’t answer you. As far as I know, migrating this extension to Python3 should not be difficult. But I am afraid there is no resources available to complete this task :(.

**Comment by FedericOldani** [comment]
Hi! I just wanna ask you @syats and @espenfl if you have found a solution to this problem. I'm interested in this too.

**Comment by espenfl** [comment]
@FedericOldani No, unfortunately not. It is still on the map of something we would like to fix, but we have not yet had the resources or time to issue a PR.

**Comment by ryangermann-gov-on-ca** [comment]
We also need OAuth2 authentication in CKAN 2.9x. We will try to make it work and provide the update, but please don't wait on us: we're still learning and hopefully we can get this working but my team is new to Python / CKAN extension development.

**Comment by frafra** [comment]
Please test and give feedback to https://github.com/conwetlab/ckanext-oauth2/pull/46.
